src/app/pages/network/configuration/configuration.component.ts

```diff
@@ -0,0 +1,325 @@
+    httpproxy: [''],
+    netwait_enabled: [false],
+    netwait_ip: [[] as string[]],
+    hosts: [[] as string[]],
+  });
+
+  readonly helptext = helptext;
+
+  hostname = {
+    fcName: 'hostname',
+    label: helptext.hostname_placeholder,
+    tooltip: helptext.hostname_tooltip,
+  };
+
+  hostname_b = {
+    fcName: 'hostname_b',
+    label: helptext.hostname_b_placeholder,
+    tooltip: helptext.hostname_b_tooltip,
+    hidden: true,
+  };
+
+  hostname_virtual = {
+    fcName: 'hostname_virtual',
+    label: helptext.hostname_virtual_placeholder,
+    tooltip: helptext.hostname_virtual_tooltip,
+    hidden: true,
+  };
+
+  domain = {
+    fcName: 'domain',
+    label: helptext.domain_placeholder,
+    tooltip: helptext.domain_tooltip,
+  };
+
+  domains = {
+    fcName: 'domains',
+    label: helptext.domains_placeholder,
+    tooltip: helptext.domains_tooltip,
+  };
+
+  netbios = {
+    fcName: 'netbios',
+    label: helptext.netbios_placeholder,
+    tooltip: helptext.netbios_tooltip,
+  };
+
+  mdns = {
+    fcName: 'mdns',
+    label: helptext.mdns_placeholder,
+    tooltip: helptext.mdns_tooltip,
+  };
+
+  wsd = {
+    fcName: 'wsd',
+    label: helptext.wsd_placeholder,
+    tooltip: helptext.wsd_tooltip,
+  };
+
+  nameserver1 = {
+    fcName: 'nameserver1',
+    label: helptext.nameserver1_placeholder,
+    tooltip: helptext.nameserver1_tooltip,
+  };
+
+  nameserver2 = {
+    fcName: 'nameserver2',
+    label: helptext.nameserver2_placeholder,
+    tooltip: helptext.nameserver2_tooltip,
+  };
+
+  nameserver3 = {
+    fcName: 'nameserver3',
+    label: helptext.nameserver3_placeholder,
+    tooltip: helptext.nameserver3_tooltip,
+  };
+
+  ipv4gateway = {
+    fcName: 'ipv4gateway',
+    label: helptext.ipv4gateway_placeholder,
+    tooltip: helptext.ipv4gateway_tooltip,
+  };
+
+  ipv6gateway = {
+    fcName: 'ipv6gateway',
+    label: helptext.ipv6gateway_placeholder,
+    tooltip: helptext.ipv6gateway_tooltip,
+  };
+
+  outbound_network_activity = {
+    fcName: 'outbound_network_activity',
+    label: '',
+    tooltip: '',
+    options: of([
+      // deny type + empty list
+      {
+        label: helptext.outbound_network_activity.allow.placeholder,
+        value: NetworkActivityType.Deny,
+        tooltip: helptext.outbound_network_activity.allow.tooltip,
+      },
+      // allow type + empty list
+      {
+        label: helptext.outbound_network_activity.deny.placeholder,
+        value: NetworkActivityType.Allow,
+        tooltip: helptext.outbound_network_activity.deny.tooltip,
+      },
+      {
+        label: helptext.outbound_network_activity.specific.placeholder,
+        value: 'SPECIFIC',
+        tooltip: helptext.outbound_network_activity.specific.tooltip,
+      },
+    ]),
+  };
+
+  outbound_network_value = {
+    fcName: 'outbound_network_value',
+    label: '',
+    tooltip: helptext.outbound_network_value.tooltip,
+    options: this.ws.call('network.configuration.activity_choices').pipe(
```

Comment:
there is `choicesToOptions()` pipe

Reply:
`network.configuration.activity_choices` returns type [string, string][], so created a new map function `arrayToOptions`